core/runtime/jsruntime.rs

```diff
@@ -670,10 +670,26 @@ impl JsRuntime {
       waker,
     });
 
-    let context_state = Rc::new(ContextState {
-      isolate: Some(isolate_ptr),
-      ..Default::default()
-    });
+    let count = ops.len();
+    let op_metrics_fns = ops
+      .iter()
+      .enumerate()
+      .map(|(id, decl)| {
+        options
+          .op_metrics_factory_fn
+          .as_ref()
+          .and_then(|f| (f)(id as _, count, decl))
+      })
+      .collect::<Vec<_>>();
+    let ops_with_metrics = op_metrics_fns
+      .iter()
+      .map(|o| o.is_some())
+      .collect::<Vec<_>>();
+    let context_state = Rc::new(ContextState::new(
+      isolate_ptr,
+      options.get_error_class_fn.unwrap_or(&|_| "Error"),
+      ops_with_metrics,
+    ));
```

Comment:
So does that mean that now metrics are always collected using the new mechanism? Could we abstract it away in `ContextState` maybe?

Reply:
Nope, this just means we keep a vec of booleans so we can look up which ops have metrics enabled. This means we no longer need to feed the `metrics_enabled` flag through to the op driver as we can check for it when polling the event loop. 

I just realized I missed something here - we are double-collecting the metrics functions. I'll fix that before landing.

Reply:
Fixed.